logstash-core/lib/logstash/runner.rb

```diff
@@ -276,6 +276,17 @@ def execute
     LogStash::PLUGIN_REGISTRY.hooks.register_emitter(self.class, @dispatcher)
 
     @settings.validate_all
+
+    if (@settings.registered?("xpack.monitoring.enabled") && @settings.get_value("xpack.monitoring.enabled") == true)
```

Comment:
I don't see a need here to only perform the deprecation check if the feature is enabled, only checking that it is registered is enough (same for management)

Reply:
ok fixed.